src/routes/transactions/helpers/transaction-verifier.helper.ts

```diff
@@ -119,6 +119,9 @@ export class TransactionVerifierHelper {
     ) {
       throw new HttpExceptionNoLog(ErrorMessage.InvalidNonce, code);
     }
+
+    this.verifyApiTransaction(args);
```

Comment:
We were trusting the API transaction as is whereas it should be verified in a similar manner as we do for the queue.